consensus/src/execution_ctx.rs

```diff
@@ -152,42 +153,117 @@ impl<'a, T: Operations + 'static, DB: Database> ExecutionCtx<'a, T, DB> {
+                                                // and terminate the round
+                                                //
+                                                // INFO: by returning here, we
+                                                // let the Consensus task
+                                                // broadcast the message
+                                                return Ok(step_result);
+                                            }
+                                            RatificationResult::Fail(vote) => {
+                                                info!(
+                                                    mode = "open_consensus",
+                                                    event =
+                                                        "ignoring Fail Quorum",
+                                                    ?vote
+                                                );
+                                            }
+                                        }
+                                    }
+
+                                    // In open consensus mode, the step is only
+                                    // terminated in case of Success Quorum.
+                                    // The acceptor will cancel the consensus if
+                                    // a block is accepted
+                                    continue;
                                 } else {
-                                    info!(
-                                        mode = "open_consensus",
-                                        event = "ignoring failed quorum",
-                                        ?vote
-                                    );
+                                    self.report_elapsed_time().await;
+                                    return Ok(step_result);
                                 }
                             }
-                            // In open consensus mode, consensus step is never
-                            // terminated.
-                            // The acceptor will cancel the consensus if a
-                            // block is accepted
-                            continue;
-                        } else {
-                            self.report_elapsed_time().await;
-                            return Ok(step_result);
+                        }
+
+                        // Handle Quorum messages from the network
+                        Payload::Quorum(qmsg) => {
+                            // We only handle messages for the current round
+                            // and branch, and iteration <= current_iteration
+                            let cur_round = self.round_update.round;
+                            let cur_prev = self.round_update.hash();
+                            let cur_iter = self.iteration;
+                            if qmsg.header.round == cur_round
```

Comment:
Same as above. Using exclusion statement first we can tame the nesting. Consider modifying to the following:

```
if qmsg.header.round != cur_round || qmsg.header.prev_block_hash != cur_prev || qmsg.header.iteration > cur_iter {
    // continue with the step execution
    continue;
}

let qiter = ...
```

Reply:
Done